codebundles/k8s-argocd-helm-health/runbook.robot

```diff
@@ -0,0 +1,98 @@
+*** Settings ***
+Documentation       This codebundle runs a series of tasks to identify potential helm release issues related to ArgoCD managed Helm objects.
+Metadata            Author    Nitin Madhok (nmadhok@icloud.com)
+Metadata            Display Name    Kubernetes ArgoCD HelmRelease TaskSet
+Metadata            Supports    Kubernetes,AKS,EKS,GKE,OpenShift,ArgoCD
+
+Library             RW.Core
+Library             RW.CLI
+Library             RW.platform
+
+Suite Setup         Suite Initialization
+
+
+*** Tasks ***
+List all available ArgoCD Helm releases
+    [Documentation]    List all ArgoCD helm releases that are visible to the kubeconfig.
+    [Tags]    argocd    helmrelease    available    list
+    ${helmreleases}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get ${RESOURCE_NAME} -n ${NAMESPACE} --context ${CONTEXT}
+    ...    target_service=${kubectl}
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${KUBECONFIG}
+    ...    render_in_commandlist=true
+    ${history}=    RW.CLI.Pop Shell History
+    RW.Core.Add Pre To Report    \n\nArgoCD Helm releases available: \n\n ${helmreleases.stdout}
+    RW.Core.Add Pre To Report    Commands Used:\n${history}
+
+Check ArgoCD Helm releases Status
+    [Documentation]    Check ArgoCD helm release Status.
+    [Tags]    argocd    helmrelease    status    health
+    ${argocd_helm_status}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} get ${RESOURCE_NAME} -n ${NAMESPACE} --context ${CONTEXT} -o=json | jq -r '.items[] | "Name:\\t\\t\\t" + .metadata.name + "\\nHealth Status:\\t\\t" + .status.health.status + "\\nSync Status:\\t\\t" + .status.sync.status + "\\nOperational State:\\t" + .status.operationState.message'
```

Comment:
Same comments as the above - we first need to ensure that the application type is helm. ArgoCD can deploy applications in many ways, helm being one of them, and in this codebundle we only want to focus on helm. 

Additionally, I don't think this task should focus on the sync status, as we have that detail with a simple command like this: 
```
$kubectl get applications  --all-namespaces -o wide 
NAMESPACE   NAME        SYNC STATUS   HEALTH STATUS   REVISION
jenkins     jenkins     Synced        Progressing     4.3.23
otel-demo   otel-demo   Synced        Healthy         0.22.1
```

Instead it is probably valuable to have a command that: 
1. Only looks at applications that are deployed with helm
2. Prints out the synced revision to the target revision

This would quickly identify issues where the desired version doesn't align with what's been installed.

Reply:
@stewartshea Agreed and updated in latest commit